src/action/payment.js

```diff
@@ -22,18 +22,42 @@ class PaymentAction {
    * @return {undefined}
    */
   listenForUrl(ipc) {
-    ipc.listen('open-url', async (event, url) => {
-      log.info('open-url', url);
-      if (!isLnUri(url)) {
-        return;
-      }
-      while (!this._store.lndReady) {
-        this._tOpenUri = await nap(100);
-      }
-      this.init();
-      this.setAddress({ address: url.replace(PREFIX_URI, '') });
-      this.checkType();
-    });
+    ipc.listen('open-url', (event, url) => this._openUrl(url));
+  }
+
+  /**
+   * Set the listener for the mobile app to handle incoming URIs
+   * containing lightning invoices.
+   * @param  {Object} Linking The expo api to handle incoming uris
+   * @return {undefined}
+   */
+  async listenForUrlMobile(Linking) {
+    Linking.addEventListener('url', ({ url }) => this._openUrl(url));
+    const url = await Linking.getInitialURL();
```

Comment:
> So `await` will only resolve when a user clicks on an invoice, right?

This used in case the app is opening via a `lightning:` uri. It always resolves. But `url` will be empty if the app was not opening by via link. See:
https://docs.expo.io/versions/latest/workflow/linking#handling-links-into-your-app

I added some unit tests for `payment.listenForUrlMobile`. Forgot before.